R/reportDuplicates.R

```diff
@@ -0,0 +1,29 @@
+#' Finds duplicates in a quitte object and warns about them per model and scenario.
+#' Also warn if variables are identical but value or units differ.
+
+#' @param mifdata object that can be converted with as.quitte
+#' @importFrom dplyr filter mutate select
+#' @return only the data that is duplicated. Has 0 rows if everything is fine
+#' @export
+reportDuplicates <- function(mifdata) {
+  mifdata <- as.quitte(mifdata)
+  duplicates <- mifdata %>% 
+    group_by(.data$model, .data$region, .data$variable, .data$period, .data$scenario) %>% 
```

Comment:
This is too strict.  `quitte` defines a minimum set of columns, but allows for others.  A common thing to do is to split scenario names into different dimensions (e.g. SSP + climate target + some restrictions), which would not get picked up by this.
```
# difference in unit and value, but additional columns
d <- data.frame(variable = "Emi", unit = "Mt CO2/yr", period = 2020,
                variant = c('A', 'B'), value = 1)
#   variable      unit period variant value
# 1      Emi Mt CO2/yr   2020       A     1
# 2      Emi Mt CO2/yr   2020       B     1

reportDuplicates(d)
# # A tibble: 2 × 8
#   model     scenario  region variable unit      period variant value
#   <fct>     <fct>     <fct>  <fct>    <fct>      <int> <chr>   <dbl>
# 1 (Missing) (Missing) GLO    Emi      Mt CO2/yr   2020 A           1
# 2 (Missing) (Missing) GLO    Emi      Mt CO2/yr   2020 B           1
# Warning message:
# In quitte::reportDuplicates(d) :
#   Duplicated data found for (Missing) (Missing):
# Variables: Emi (Mt CO2/yr)
```
Do
```
  group_by(!!!syms(setdiff(colnames(mifdata), c('unit', 'value'))))
```
instead.